runelite-client/src/main/java/net/runelite/client/plugins/mta/EnchantmentRoom.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2018, Jasper Ketelaar <Jasper0781@gmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.mta;
+
+import java.awt.image.BufferedImage;
+import javax.inject.Inject;
+import com.google.common.eventbus.Subscribe;
+import net.runelite.api.Client;
+import net.runelite.api.Item;
+import net.runelite.api.ItemID;
+import net.runelite.api.Player;
+import net.runelite.api.Region;
+import net.runelite.api.Tile;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.events.ConfigChanged;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetID;
+import net.runelite.api.widgets.WidgetInfo;
+
+public class EnchantmentRoom extends MTARoom
+{
+
+	private static final int IMAGE_START_ID = 4;
+	private static final int IMAGE_END_ID = 12;
+
+	private final MTAPlugin plugin;
+	private final Client client;
+
+	private BufferedImage image;
+
+	@Inject
+	public EnchantmentRoom(MTAConfig config, MTAPlugin plugin, Client client)
+	{
+		super(config);
+		this.plugin = plugin;
+		this.client = client;
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+
+		if (!inside() || !getConfig().enchantment())
+		{
+			return;
+		}
+
+
+		if (getConfig().enchantment())
+		{
+			setHiddenWidgets(true);
+		}
+
+		Region region = client.getRegion();
+		Tile[][][] tiles = region.getTiles();
+		int z = client.getPlane();
+		int count = 0;
+		Player local = client.getLocalPlayer();
+		WorldPoint nearest = null;
+		double min = Double.MAX_VALUE;
+
+		for (int x = 0; x < tiles[z].length; x++)
```

Comment:
This should use the item layer changed event probably to find/store dragonstones